cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -88,8 +89,8 @@ controller, and serviceaccounts controller.`,
 				fmt.Fprintf(os.Stderr, "%v\n", err)
 				os.Exit(1)
 			}
-
-			if err := Run(c.Complete()); err != nil {
```

Comment:
Prefer putting a Stop channel in the options struct and populating it in Complete() if nil

Reply:
make sense

Reply:
I find apiserver also pass stop channel to `Run`